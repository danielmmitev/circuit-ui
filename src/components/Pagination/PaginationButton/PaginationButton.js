import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Button from '../../Button';

const PageButtons = styled(Button)`
  ${({ theme }) => `
    margin: ${theme.spacings.kilo};
  `};
`;

const PaginationButton = ({ value, onClick, plain }) => {
  const size = plain ? Button.MEGA : Button.KILO;
  return (
    <PageButtons
      name={value}
      size={size}
      plain={plain}
      onClick={() => onClick(value)}
    >
      {value}
    </PageButtons>
  );
};

PaginationButton.propTypes = {
  value: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  plain: PropTypes.bool
};

PaginationButton.defaultProps = {
  plain: false
};

export default PaginationButton;
