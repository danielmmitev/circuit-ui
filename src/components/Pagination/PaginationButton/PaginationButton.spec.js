import React from 'react';

import PaginationButton from './PaginationButton';

describe('PaginationButton', () => {
  describe('styles', () => {
    it('should render with default styles', () => {
      const actual = shallow(
        <PaginationButton value={1} onClick={jest.fn()} />
      );
      expect(actual).toMatchSnapshot();
    });

    it('should render with default styles when have plain false', () => {
      const actual = shallow(
        <PaginationButton value={1} onClick={jest.fn()} plain />
      );
      expect(actual).toMatchSnapshot();
    });
  });

  describe('business logic', () => {
    it('should call method onClick with value passed from props', () => {
      const props = {
        value: 13,
        onClick: jest.fn()
      };
      const component = shallow(<PaginationButton {...props} />);
      component.simulate('click');
      expect(props.onClick).toHaveBeenCalledWith(13);
    });
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(
        <PaginationButton value={1} onClick={jest.fn()} />
      );
      const actual = await axe(wrapper);
      expect(actual).toHaveNoViolations();
    });
  });
});
