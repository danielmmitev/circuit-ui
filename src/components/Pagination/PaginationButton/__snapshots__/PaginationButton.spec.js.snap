// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationButton styles should render with default styles 1`] = `
<PageButtons
  disabled={false}
  flat={false}
  href={null}
  name={1}
  onClick={[Function]}
  plain={false}
  primary={false}
  secondary={false}
  size="kilo"
  stretch={false}
  target={null}
>
  1
</PageButtons>
`;

exports[`PaginationButton styles should render with default styles when have plain false 1`] = `
<PageButtons
  disabled={false}
  flat={false}
  href={null}
  name={1}
  onClick={[Function]}
  plain={true}
  primary={false}
  secondary={false}
  size="mega"
  stretch={false}
  target={null}
>
  1
</PageButtons>
`;
